### DISCUSSION TITLE
error: cannot declare variable 'player' to be of abstract type 'audio_tools::AudioPlayer'

### DISCUSSION THREAD

**Discussion by mluijben** [pschatzmann/arduino-audio-tools#375] [Q&A] [answered]
Hi,

I purchased an ESP32 Audiokit and was playing around with the examples from the audio tools.
Love the work you have done, makes it easy to work with audio on an Esp32.

The bluetooth sketches work fine but when I use any sketches with the AudioPlayer I get after 'Verify'

'error: cannot declare variable 'player' to be of abstract type 'audio_tools::AudioPlayer'

I used the example from the 'examples-audiokit'
player-url_icy-audiokit.ino

Do you know what I'm doing wrong?
Following is a copy from the example sketch and where the sketch stops with Verify

            #include "AudioTools.h"
            #include "AudioCodecs/CodecMP3Helix.h"
            #include "AudioLibs/AudioKit.h"
            
            const char *urls[] = {
              "http://stream.srg-ssr.ch/m/rsj/mp3_128",
              "http://stream.srg-ssr.ch/m/drs3/mp3_128",
              "http://stream.srg-ssr.ch/m/rr/mp3_128",
              "http://sunshineradio.ice.infomaniak.ch/sunshineradio-128.mp3",
              "http://streaming.swisstxt.ch/m/drsvirus/mp3_128"
            };
            const char *wifi = "WLANNR11";
            const char *password = "wirelessnr11";
            
            ICYStream urlStream(wifi, password);
            AudioSourceURL source(urlStream, urls, "audio/mp3");
            AudioKitStream kit;
            MP3DecoderHelix decoder;
    ---> AudioPlayer player(source,kit,decoder);

**Comment by pschatzmann** [comment, marked as answer]
I can't reproduce your issue with the latest version of the library.
I suggest that you take the latest version...

**Reply by mluijben** [reply]
Not sure what I did wrong...
Updated to the latest version and now all compiles.

Thank you for the very fast response!
